app/src/test/java/org/mozilla/fenix/utils/SettingsTest.kt

```diff
@@ -0,0 +1,268 @@
+/*
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.mozilla.fenix.utils
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.feature.sitepermissions.SitePermissionsRules
+import mozilla.components.feature.sitepermissions.SitePermissionsRules.Action.ASK_TO_ALLOW
+import mozilla.components.feature.sitepermissions.SitePermissionsRules.Action.BLOCKED
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.TestApplication
+import org.mozilla.fenix.ext.clearAndCommit
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@Config(application = TestApplication::class)
+class SettingsTest {
+
+    lateinit var settings: Settings
+
+    @Before
+    fun setUp() {
+        settings = Settings.getInstance(testContext)
+            .apply(Settings::clear)
+    }
+
+    @Test
+    fun usePrivateMode() {
+        assertFalse(settings.usePrivateMode)
+
+        settings.setPrivateMode(true)
+        assertTrue(settings.usePrivateMode)
+
+        settings.setPrivateMode(false)
+        assertFalse(settings.usePrivateMode)
```

Comment:
This API is pretty grim/Java-y. This should not be fixed in this PR :)

Reply:
Do you mean api of the component or test style?

Reply:
Anyway, I usually use `Given-When-Then` comments to improve readability. Will add 'em here as well.